### PR TITLE
[NFC] W/A GCC maybe-uninitialized warning

### PR DESCRIPTION
**Pull request by MrSidims**

### CODE REVIEW

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -4385,9 +4385,12 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
       Extracts[Idx] = BM->addVectorExtractDynamicInst(
           VecSVal, BM->addIntegerConstant(I32STy, Idx), BB);
     }
+    assert(Extracts[0] && "Uninitialized Extracts of vector reduce lowering");
```
[the remark this answers is not in the document]
Comment (svenvh) [reply lib/SPIRV/SPIRVWriter.cpp]: Best to check that `VecSize > 0` indeed, but we have to do that with an `if`; assertions won't help in release builds.

**Comment by MrSidims** [comment]
New patch new GCC warning. @svenvh I expect even more as for out-of-tree build of LLVM GCC won't see LLVM sources to make a correct judgment. Would it makes sense to disable this warning? On the other hand fixing them all would enhance the code base in a sense of not trusting LLVM's API outputs.

**Comment by svenvh** [comment]
I'm quite surprised why this is suddenly happening; both the failing builds and the last passing build report "The CXX compiler identification is GNU 9.4.0".

If it's not too much effort, it would be nice to fix all of the warnings.  On your latest patch the build made it to 88% so hopefully it's not much more to fix.

**Comment by MrSidims** [comment]
> I'm quite surprised why this is suddenly happening; both the failing builds and the last passing build report "The CXX compiler identification is GNU 9.4.0".
> 
> If it's not too much effort, it would be nice to fix all of the warnings. On your latest patch the build made it to 88% so hopefully it's not much more to fix.

Me too. Locally I'm building out-of-tree as well, but I don't see any warnings.

**Comment by svenvh** [comment]
> Me too. Locally I'm building out-of-tree as well, but I don't see any warnings.

I can reproduce something locally at least, with GCC 9.4.0 and `-DCMAKE_BUILD_TYPE=Release`.  I can't explain the recent regressions though.

Funny enough I'm only getting the DenseMap warnings; not the warning that you're fixing with this PR.

**Comment by svenvh** [comment]
Perhaps it's an issue with older gcc versions.  Let's see what happens with https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2589

**Pull request merged**